feature_flags_test.go

```diff
@@ -0,0 +1,2882 @@
+			w.Write([]byte(fixture("test-decide-v2.json")))
+		} else if strings.HasPrefix(r.URL.Path, "/api/feature_flag/local_evaluation") {
+			w.Write([]byte(fixture("feature_flag/test-compute-inactive-flags-locally-2.json")))
+		}
+	}))
+
+	defer server.Close()
+
+	client, _ = NewWithConfig("Csyjlnlun3OzyNJAafdlv", Config{
+		PersonalApiKey: "some very secret key",
+		Endpoint:       server.URL,
+	})
+	defer client.Close()
+
+	featureVariants, _ = client.GetAllFlags(FeatureFlagPayloadNoKey{
+		DistinctId: "distinct-id",
+	})
+
+	if featureVariants["beta-feature"] != false || featureVariants["disabled-feature"] != true {
+		t.Error("Should match")
+	}
+}
+
+func TestFeatureEnabledSimpleIsTrueWhenRolloutUndefined(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(fixture("feature_flag/test-simple-flag-without-rollout.json")))
+	}))
+	defer server.Close()
+
+	client, _ := NewWithConfig("Csyjlnlun3OzyNJAafdlv", Config{
+		PersonalApiKey: "some very secret key",
+		Endpoint:       server.URL,
+	})
+	defer client.Close()
+
+	isMatch, _ := client.IsFeatureEnabled(
+		FeatureFlagPayload{
+			Key:        "simple-flag",
+			DistinctId: "distinct-id",
+		},
+	)
+	if !isMatch {
+		t.Error("Should be enabled")
+	}
+}
+
+func TestGetFeatureFlag(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.HasPrefix(r.URL.Path, "/decide") {
+			w.Write([]byte(fixture("test-decide-v2.json")))
+		} else if strings.HasPrefix(r.URL.Path, "/api/feature_flag/local_evaluation") {
+			w.Write([]byte(fixture("feature_flag/test-simple-flag-person-prop.json")))
+		}
+	}))
+
+	defer server.Close()
+
+	client, _ := NewWithConfig("Csyjlnlun3OzyNJAafdlv", Config{
+		PersonalApiKey: "some very secret key",
+		Endpoint:       server.URL,
+	})
+	defer client.Close()
+
+	variant, _ := client.GetFeatureFlag(
+		FeatureFlagPayload{
+			Key:        "test-get-feature",
+			DistinctId: "distinct_id",
+		},
+	)
+
+	if variant != "variant-1" {
+		t.Error("Should match")
+	}
+}
+
+func TestCaptureIsCalled(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.HasPrefix(r.URL.Path, "/decide") {
+			w.Write([]byte(fixture("test-decide-v2.json")))
+		} else if strings.HasPrefix(r.URL.Path, "/api/feature_flag/local_evaluation") {
+			w.Write([]byte(fixture("feature_flag/test-simple-flag-person-prop.json")))
+		}
+	}))
+
+	defer server.Close()
+
+	client, _ := NewWithConfig("Csyjlnlun3OzyNJAafdlv", Config{
+		PersonalApiKey: "some very secret key",
+		Endpoint:       server.URL,
+	})
+	defer client.Close()
+
+	variant, _ := client.GetFeatureFlag(
+
+		FeatureFlagPayload{
+			Key:        "test-get-feature",
+			DistinctId: "distinct_id",
+		},
+	)
+
+	if variant != "variant-1" {
+		t.Error("Should match")
+	}
+
+}
+
+func TestCaptureMultipleUsersDoesntOutOfMemory(t *testing.T) {
```

Comment:
I'm unsure how to test this

Reply:
Not aware of the intricacies here, but the test is testing that the size of the hash doesn't blow up arbitrarily with new users coming in. (As for long running server-side apps, there's a risk of this running out of memory).

In node, I just check that the distinctIdHasSentFeatureFlagCalls array never exceeds MAX_SIZE, when 100*MAX_SIZE new users are seen:

https://github.com/PostHog/posthog-js-lite/pull/14/files#diff-f8fbefb57e174bbb58200dfeab663d6b65aa93b671b7847f4f48a17bfdf5dd54R272

Reply:
I'm testing the sizelimitedmap separately as a unit test. Doesn't seem feasible to test as an integration test since there's no way to expose the map without making it part of the public interface on client which isn't good practice as it's unnecessary to the user

Reply:
Works 👍